server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -343,7 +343,9 @@ public static GetResult extractGetResult(
         }
         BytesReference sourceFilteredAsBytes = sourceAsBytes;
         if (request.fetchSource().hasFilter()) {
-            sourceFilteredAsBytes = Source.fromMap(source, sourceContentType).filter(request.fetchSource().filter()).internalSourceRef();
+            sourceFilteredAsBytes = Source.fromMap(source, sourceContentType)
+                .filter(request.fetchSource().filter(null))
+                .internalSourceRef();
```

Comment:
We need to investigate if it ever makes sense to exclude vectors in an update request. My brief manual testing indicates no, but let's dig a bit deeper.

If we don't want users to exclude vectors in an update request, we need to change update request parsing such that `include_vectors` cannot be parsed in that context.

Reply:
I had a chance to look into the `update_by_query` [[API docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html)] and `update_api` [[API docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html)] functionality. 

If I understood correctly, `update_by_field` does not have access to the `_source` field, and `update_api` has access to the `_source` field. However, I think in both cases we should return vector fields by default as these APIs can be used to update [field mapping](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html#docs-update-by-query-api-example).

Reply:
I think we only ever want to exclude vectors in search results. If we exclude them in update requests I believe there will be side effects that we don't want.

Reply:
I did a deep dive into the Update API to figure out what's going on with source filtering. 
TL;DR: The [documentation, stating that _source is true by default,](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#docs-update-api-query-params) is wrong 🫠 

What actually happens is that when `_source` is not specified, [UpdateRequest.fetchSourceContext](https://github.com/elastic/elasticsearch/blob/1fafdb1da0034d3e69163f13f141a7bf99ca923f/server/src/main/java/org/elasticsearch/action/update/UpdateRequest.java#L108) is null. Later in the response building process, [UpdateHelper interprets this null value as equivalent to _source == false](https://github.com/elastic/elasticsearch/blob/6ca3ac253a8d8171f228d9dbfd6c5c924239c226/server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java#L341). And thus, by default source retrieval is disabled!

Here's a test case to prove it:

```json
PUT test-index

PUT test-index/_doc/1
{
  "key1": "value1"
}

POST test-index/_update/1
{
  "doc": {
    "key2": "value2"
  }
}
```

```json
{
    "_index": "test-index",
    "_id": "1",
    "_version": 10,
    "result": "updated",
    "_shards": {
        "total": 2,
        "successful": 1,
        "failed": 0
    },
    "_seq_no": 9,
    "_primary_term": 1
}
```

```json
POST test-index/_update/1?_source=true
{
  "doc": {
    "key2": "value3"
  }
}
```

```json
{
    "_index": "test-index",
    "_id": "1",
    "_version": 11,
    "result": "updated",
    "_shards": {
        "total": 2,
        "successful": 1,
        "failed": 0
    },
    "_seq_no": 10,
    "_primary_term": 1,
    "get": {
        "_seq_no": 10,
        "_primary_term": 1,
        "found": true,
        "_source": {
            "key1": "value1",
            "key2": "value3"
        }
    }
}
```

I also traced the Update API code path and `_source` is used exclusively for filtering the source returned to the user. It has no effect on how the document is actually updated.

What does this all mean? We need to pass mappings to the `filter` call, because by default we should filter out embeddings from the source if source retrieval is enabled. This should not be a problem, we can do this by adding an `IndicesService` to `UpdateHelper`'s constructor and slightly changing when it is created in `NodeConstruction` :)

Reply:
I created this PR to pass mappings to update helper to enable source filtering: https://github.com/elastic/elasticsearch/pull/112839